Bottom: dede7d613dc3f45438d63b97c0ce083378cd747f
Top:    d7705924313f5cf530b3fc17441d42ee7654f487
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-08 01:57:58 +0000

Workaround crash where contour is passed as NULL to poly_PreContour











---

diff --git a/src/polygon.c b/src/polygon.c
index c0aa76d..181a6c3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
